src/UI/UIView.cpp

```diff
@@ -45,596 +54,905 @@ void UIView::draw_ui_callback()
+              m_trn_fct_options_window = false;
+              m_current_save_modal = SAVE_MODAL::SAVE_NONE;
+              ImGui::CloseCurrentPopup();
+            }
+            ImGui::SameLine();
+            if (ImGui::Button("Cancel"))
+            {
+              m_trn_fct_options_window = false;
+              m_current_save_modal = SAVE_MODAL::SAVE_NONE;
               ImGui::CloseCurrentPopup();
             }
+
             ImGui::EndPopup();
           }
         }
 
 
-      }
 
-      if (m_trn_fct_opitions_window)
-      {
-        ImGui::OpenPopup("Transfer Function Options");
-        ImGui::SetNextWindowSize(ImVec2(350, 400), ImGuiCond_FirstUseEver);
-        if (ImGui::BeginPopupModal("Transfer Function Options", &m_trn_fct_opitions_window))
+        ImGui::Checkbox("use transferfunction", &m_use_transferfunction);
+        if (m_use_transferfunction)
         {
-          ImGui::Text("Change Name");
-          char* writable = new char[m_copy_trnfnct_name.size() + 1];
-          std::copy(m_copy_trnfnct_name.begin(), m_copy_trnfnct_name.end(), writable);
-          writable[m_copy_trnfnct_name.size()] = '\0'; // don't forget the terminating 0
-          //ImGui::PushItemWidth(-1);
-          ImGui::InputText("##text1", writable, IM_ARRAYSIZE(writable));
-          ImGui::IsItemActive();
-          //ImGui::PopItemWidth();
-
-          ImGui::Separator();
-          /*float q_min = 0;
-          float q_max = 0;
-          tfn_widget[m_table_selection].get_Quantiles(q_min, q_max);
-          m_histogram_quantiles[0] = q_min;
-          m_histogram_quantiles[1] = q_max;*/
-
-          ImGui::Text("Quantiles");
-          ImGui::SliderFloat2("##Quantiles", m_histogram_quantiles, 0.05f, 0.95f);
-
-          if (ImGui::Button("Adjust to Histogram"))
+
+
+          bool is_multi_channel = m_controller_app.data_is_multi_channel();
+          if (is_multi_channel)
           {
-            adjust_transfer_function_to_histogram();
-            tfn_widget[m_table_selection].alpha_control_pts.clear();
-            tfn_widget[m_table_selection].alpha_control_pts.push_back(vec2f(0.0, 0.0));
-            tfn_widget[m_table_selection].alpha_control_pts.push_back(m_histogram_point_1);
-            tfn_widget[m_table_selection].alpha_control_pts.push_back(m_histogram_point_2);
-            tfn_widget[m_table_selection].alpha_control_pts.push_back(vec2f(1.0, 1.0));
-            tfn_widget[m_table_selection].set_Quantiles(m_histogram_quantiles[0], m_histogram_quantiles[1]);
-            ImGui::OpenPopup("Confirmation");
-            ImGui::SetNextWindowSize(ImVec2(350, 400), ImGuiCond_FirstUseEver);
-          }
+            for (int i = 0; i < 3; i++) {
+              if (m_animated)
+              {
+                /*
+                  frame by frame animation
+
+                  unsigned int active_volume = floor(m_frame);
+                  unsigned int active_volume2 = ceil(m_frame);
```

Comment:
delete commented out code

Reply:
Let me keep this logic. I still take it as a base to fix the volume animation frame rate.

Reply:
as long as you add a `TODO` comment including an issue number that indicates what the fix is for (so also adding an issue for what needs to be done)

Reply:
I added the `TODO` comment and removed unused code.